### PR TITLE
Add snappy compression

### PR DESCRIPTION
**Pull request by shtelzerartem**
## What does this PR do?

Adds Snappy compression adapter

## Test Plan

Added 2 tests for Snappy compression similarly to ZSTD, both passed tests

Screen Shot 2022-10-08 at 18 48 53
Screen Shot 2022-10-08 at 18 48 37

 - [x] `testCreateBucketFileSnappyCompression`
 - [x] `testFilePreviewSnappyCompression`

## Related PRs and Issues

Resolves [#4002](https://github.com/appwrite/appwrite/issues/4002)
Relates to my second PR [#4002-utopia-php](https://github.com/utopia-php/storage/pull/56)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yep :)

### CODE REVIEW

**Comment by Meldiron** [comment]
@shtelzerartem , thanks for the PR! 🤯 Please give us some time to review it.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Looks good! I'm going to add the `hacktoberfest-accepted` label, but the core team will need to clean up some things before this can be merged:

- [ ] Update API specs
- [ ] Bump utopia-php/storage in composer.json
- [ ] Fix merge conflict

File: composer.json
Comment (christyjacob4) [inline composer.json:83]: reminder to remove this

File: app/views/console/storage/bucket.phtml
Comment (christyjacob4) [inline app/views/console/storage/bucket.phtml:11]: We need to remove this file. We no longer use it. You need to add the compression option here
https://github.com/appwrite/console/blob/72b3f26edb802517b8d1c9d02900dbb99da58d38/src/routes/console/project-%5Bproject%5D/storage/bucket-%5Bbucket%5D/settings/+page.svelte#L49

File: app/controllers/api/storage.php
Comment (christyjacob4) [inline app/controllers/api/storage.php:69]: Please replace the string literals with the constants in all occurances in this file

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by christyjacob4** [comment]
@shtelzerartem thanks a lot for your contributions during Hacktoberfest 2022! 

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord. 

I'm closing this PR as there has not been any activity on this for quite a while. If you'd like to continue working on it , feel free to create a new PR.

**Pull request closed without merging**
